mono/metadata/locales.c

```diff
@@ -530,22 +532,40 @@ get_current_locale_name (void)
 	return ret;
 }
 
-MonoString*
-ves_icall_System_Globalization_CultureInfo_get_current_locale_name (void)
+static void
+culture_info_get_current_locale_name (MonoStringHandle ret, MonoError *error)
```

Comment:
I really don't like the idea of passing in the return handle that will then be modified.

It leads to some ugly code.